_posts/2017-10-19-haskell-learning-group-2nd-meeting.md

```diff
@@ -0,0 +1,75 @@
+
+## Tooling
+At this point we don't need to know or to use many different tools, so REPL was good to start with. REPL stands for **r**ead-**e**val-**p**rint **l**oop, an interactive programming shell that evaluates expressions and returns results in the same environment. For Haskell this is [GHCi][ghci]. One way of getting it is by installing [stack][stack]. Stack also provides many other things that we will be using in the future. It can be used for installing packages (including dependencies to specific project), and also building, testing and benchmarking your project.
+For now we will be using it only for REPL and running tests later on. It can be started in console by running `ghci`. We began to play there by executing simple expressions and see results. Sometimes its more convenient to write code in an editor. Its possible to save source code in lets say `test.hs` file and load it as a module later on by running `:load filename.hs` in ghci.
+
+[ghci]: https://docs.haskellstack.org/en/stable/ghci/
+[stack]: https://docs.haskellstack.org/en/stable/README/
+
+## Hello World
+
+It all starts with printing out the "Hello World!" and Haskell is not an exception. Here is how this simple program can look like in Haskell:
+```haskell
+sayHello :: String -> IO ()
+sayHello x = putStrLn ("Hello, " ++ x ++ "!")
+```
+
+Well, I guess just ouputting a string was too easy, so we started with a bit more upgraded version of the classic program. Instead of hardcoding "World", this function takes one argument which is a string and says "Hello" to whatever was passed as a parameter to it. 
+We went briefly through the the code, looked at type definition and try to run it to see results. After loading this module from `hello.hs`, we got obvious results by executing `sayHello "World"`.  
+
+## Basic Expressions and Operations
+
+During the session we quickly went through arithmetical operations. Nobody was having any real issues with the exercises or understanding how it works so we went through all of them rather quickly. We spend some time writing expressions in infix style. So normal way of writing it `1 + 2`, with infix style would look like this `(+) 1 2`. There is a useful command in ghci which calls `:info`, we started to use it for different functions we wanted to get information about. It is not limited to functions and also works for operations like multiply or division: `:info *`. It's useful to get type information and also to find out if its infix operator. For `*` it says `infixl 7 *`. We learned how to read and understand provided info. So it's an infix operator, left associative with precedence `7`.
+Something totally new for us was `$`, which has precedence `0`. It is an infix operator and it can be convenient for reducing amount of parentheses in expression. We tried to experiment with it and had a look on how it might be used. For instance, simple example:
```

Comment:
amount -> the amount
expression -> an expression
might be used -> could be used
For instance, simple example -> For example